src/Microsoft.AspNetCore.Routing/EndpointNameAddressScheme.cs

```diff
@@ -15,7 +15,13 @@ internal class EndpointNameAddressScheme : IEndpointAddressScheme<string>
 
         public EndpointNameAddressScheme(CompositeEndpointDataSource dataSource)
         {
-            _cache = new DataSourceDependentCache<Dictionary<string, Endpoint[]>>(dataSource, Initialize);
+            // Do not use method group to avoid extra allocation
+            // Can be removed once this is improved in either:
+            // Roslyn: https://github.com/dotnet/roslyn/issues/5835
+            // -or-
+            // JIT: https://github.com/dotnet/coreclr/issues/1784
+            _cache = new DataSourceDependentCache<Dictionary<string, Endpoint[]>>(dataSource,
+                                                                                  (endpoints) => Initialize(endpoints));
```

Comment:
Is this an extra allocation each time the cache is used, or an extra allocation when EndpointNameAddressScheme is created? If it is only one allocation when EndpointNameAddressScheme is created then it isn't worth changing - this class is a singleton.

Reply:
The improvement is so minor, and it is one the jit/clr will eventually make anyway, that it isn't worth it. Thanks for the suggestion tho!